firmware/library/L1_Drivers/uart.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <cstdint>
+
+class UART_Interface
+{
+ public:
+    virtual void SetBaud(uint32_t baudrate) = 0;
+    virtual bool Initialize(uint32_t baud, uint32_t queue_size = 16) = 0;
```

Comment:
You should also provide methods for setting queue_size.

Reply:
You stated during the design doc phase that you didn't want the queue_size to be dynamically set. This is why I didn't provide a method to set the queue size before. Did you want to have queue_size dynamically set now?

Reply:
My original thought was that you were creating a queue, but now I see this refers to the FIFO (correct me if I'm wrong). Also, looking at the datasheet, I don't see a register that allows you to change the FIFO length. If that is the case, then remove queue_size as an option.

Reply:
What if the queue_size parameter is used for a FreeRTOS queue being initialized in the function?